### PR TITLE
Add timezone to formatISO

### PR DESCRIPTION
**Pull request by Imballinst**
Fixes https://github.com/date-fns/date-fns/issues/1559.

Signed-off-by: Try Ajitiono <ballinst@gmail.com>

### CODE REVIEW

File: src/formatISO/test.js [hunk 1]
```diff
@@ -3,25 +3,54 @@
...
+    const sign = offset < 0 ? '+' : '-'
+
+    tzOffset = `${sign}${hourOffset}:${minuteOffset}`
+  } else {
+    // The notation "Z" only applies for basic format AND if the timezone offset is 0.
+    tzOffset = 'Z'
+  }
+
+  return tzOffset
+}
```
Comment (Imballinst) [inline src/formatISO/test.js]: This is similar with https://github.com/date-fns/date-fns/blob/fa96c90128e87024f7e8f344e4f764924ebcc542/src/formatRFC3339/test.js#L8-L26, but different logic in the first `if` branch.

In the RFC format, there is no `extended` format, while ISO formats have it.

File: src/formatISO/index.js [hunk 2]
```diff
@@ -95,7 +112,7 @@ export default function formatISO(dirtyDate, dirtyOptions) {
     const separator = result === '' ? '' : 'T'
 
     // HHmmss or HH:mm:ss.
-    result = `${result}${separator}${hour}${timeDelimiter}${minute}${timeDelimiter}${second}`
+    result = `${result}${separator}${hour}${timeDelimiter}${minute}${timeDelimiter}${second}${tzOffset}`
```
Comment (Imballinst) [inline src/formatISO/index.js]: now that I think about it, is it better to do `[hour, minute, second].join(timeDelimiter)` instead for readability purposes? 😅

File: src/formatISO/test.js [hunk 3]
```diff
@@ -3,25 +3,54 @@
...
+    const sign = tzOffset < 0 ? '+' : '-'
+
+    offset = `${sign}${hourOffset}:${minuteOffset}`
+  } else {
+    // The notation "Z" only applies for basic format AND if the timezone offset is 0.
+    offset = 'Z'
+  }
+
+  return offset
+}
```
Comment (Imballinst) [inline src/formatISO/test.js]: This is similar with https://github.com/date-fns/date-fns/blob/fa96c90128e87024f7e8f344e4f764924ebcc542/src/formatRFC3339/test.js#L8-L26, but different logic in the first `if` branch.

In the RFC format, there is no `extended` format, while ISO formats have it.

**Comment by Imballinst** [comment]
I'll fix the CI test slightly later™

File: src/formatISO/index.js (hunk 2 above)
Comment (kossnocorp) [reply src/formatISO/index.js]: @Imballinst yes, that makes sense 👍

File: src/formatISO/test.js [hunk 4]
```diff
@@ -34,12 +62,16 @@ describe('formatISO', () => {
 
   it('formats only time', () => {
     const date = new Date(2019, 2 /* Mar */, 3, 19, 0, 52, 123)
+    const tzOffsetBasic = generateOffset(date)
+    const tzOffsetExtended = generateOffset(date)
```
Comment (kossnocorp) [inline src/formatISO/test.js]: I suppose these two are equal, so why don't we use a single variable?

Comment (Imballinst) [reply src/formatISO/test.js]: ...right. I'll update it.

File: src/formatISO/index.js [hunk 5]
```diff
@@ -12,7 +12,7 @@ import addLeadingZeros from '../_lib/addLeadingZeros/index.js'
  *
  * @param {Date|Number} date - the original date
  * @param {Object} [options] - an object with options.
- * @param {'extended'|'basic'} [options.format='extended'] - if 'basic', hide delimiters between date and time values.
+ * @param {'extended'|'basic'} [options.format='extended'] - if 'basic', hide delimiters between date and time values and hide time zone.
```
Comment (Imballinst) [inline src/formatISO/index.js]: @kossnocorp I have updated the JSDoc comments. Please take a look!

**Comment by kossnocorp** [comment]
@Imballinst please hold on, the review is coming. I don't like the fact that timezone presence is tied to delimiters, because they aren't connected. But I can't find a solution yet, and without a solution, the review is pointless.

I hope @leshakoss can help me here because it was he who developed the current argument schema.

**Comment by Imballinst** [comment]
> @Imballinst please hold on, the review is coming. I don't like the fact that timezone presence is tied to delimiters, because they aren't connected. But I can't find a solution yet, and without a solution, the review is pointless.
> 
> I hope @leshakoss can help me here because it was he who developed the current argument schema.

Sorry for that! I thought you missed this. Thanks for clarifying!

Could you tell more about "timezone tied to delimiters"? Isn't the timezone only tied to the `time` and `complete` representation (regardless of `basic` or `extended` format)?

Alternatively, we could add something like, `includeTimezone` field in the `options` object. But yeah, I agree, this needs to be discussed. Perhaps there is a best way to do that.

**Comment by kossnocorp** [comment]
> Could you tell more about "timezone tied to delimiters"? Isn't the timezone only tied to the time and complete representation (regardless of basic or extended format)?

You're right I completely missed it. I'm not even sure what I was talking about lol. Sorry about that.

I took a second look at the code and the current behavior is what we need.

Sorry again.

File: src/formatISO/index.js (hunk 5 above)
Comment (kossnocorp) [reply src/formatISO/index.js]: Oh god damn, that's the line that confused me. The docs are just misleading. I'll fix that in the master.

**Pull request merged**